diffrax/_integrate.py

```diff
@@ -1006,29 +1012,35 @@ def _promote(yi):
     del timelikes
 
     # Backward compatibility
-    if isinstance(
-        solver, (EulerHeun, ItoMilstein, StratonovichMilstein)
-    ) and _term_compatible(
-        y0, args, terms, (ODETerm, AbstractTerm), solver.term_compatible_contr_kwargs
-    ):
-        warnings.warn(
-            "Passing `terms=(ODETerm(...), SomeOtherTerm(...))` to "
-            f"{solver.__class__.__name__} is deprecated in favour of "
-            "`terms=MultiTerm(ODETerm(...), SomeOtherTerm(...))`. This means that "
-            "the same terms can now be passed used for both general and SDE-specific "
-            "solvers!",
-            stacklevel=2,
-        )
-        terms = MultiTerm(*terms)
+    if isinstance(solver, (EulerHeun, ItoMilstein, StratonovichMilstein)):
+        try:
+            _assert_term_compatible(
+                y0,
+                args,
+                terms,
+                (ODETerm, AbstractTerm),
+                solver.term_compatible_contr_kwargs,
+            )
+            warnings.warn(
+                "Passing `terms=(ODETerm(...), SomeOtherTerm(...))` to "
+                f"{solver.__class__.__name__} is deprecated in favour of "
+                "`terms=MultiTerm(ODETerm(...), SomeOtherTerm(...))`. This means that "
+                "the same terms can now be passed used for both general "
+                "and SDE-specific solvers!",
+                stacklevel=2,
+            )
+            terms = MultiTerm(*terms)
+        except Exception as _:
+            pass
```

Comment:
This would be better structured as
```python
try:
    _assert_term_compatible(...)
except ValueError:
    pass
else:
    warnings.warn(...)
    terms = MultiTerm(*terms)
```
to avoid placing too much in the `try` block.